src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -37,52 +44,210 @@ public sealed class SonarAnalyzerManager
         internal /*for testing purposes*/ static readonly Version AnalyzerVersion = AnalyzerAssemblyName.Version;
         internal /*for testing purposes*/ static readonly string AnalyzerName = AnalyzerAssemblyName.Name;
 
-        internal /*for testing purposes*/ SonarAnalyzerManager(IActiveSolutionBoundTracker activeSolutionBoundTracker,
-            Workspace workspace)
+        private DelegateInjector delegateInjector;
+        private ISonarQubeIssuesProvider sonarqubeIssueProvider;
+        private SuppressionHandler suppressionHandler;
+
+        public SonarAnalyzerManager(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
+
+            this.activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
+            Debug.Assert(this.activeSolutionBoundTracker != null, "Cannot retrieve an object of type " +
+                $"{nameof(IActiveSolutionBoundTracker)} from MEF catalog.");
+            this.activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            this.logger = serviceProvider.GetMefService<ILogger>();
+            Debug.Assert(this.logger != null, $"Cannot retrieve an object of type {nameof(ILogger)} from MEF catalog.");
+
+            var componentModel = serviceProvider.GetService<SComponentModel, IComponentModel>();
+            this.workspace = componentModel.GetService<VisualStudioWorkspace>();
+            Debug.Assert(this.workspace != null, $"Cannot retrieve an object of type {nameof(VisualStudioWorkspace)}.");
+
+            this.sonarQubeService = serviceProvider.GetMefService<ISonarQubeService>();
+            Debug.Assert(this.sonarQubeService != null, $"Cannot retrieve an object of type {nameof(ISonarQubeService)} " +
+                "from MEF catalog.");
+
+            SetupVsixSonarAnalyzerDelegates();
+        }
+
+        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
         {
-            if (activeSolutionBoundTracker == null)
+            RefreshSuppresionHandlingInNuGetAnalyzers();
+        }
+
+        #region NuGet SonarAnalyzer Delegates
+        private void RefreshSuppresionHandlingInNuGetAnalyzers()
+        {
+            // This method can be called on the UI thread so unhandled exceptions will crash VS
+            try
+            {
+                if (activeSolutionBoundTracker.CurrentBindingConfiguration.Mode != NewConnectedMode.SonarLintMode.Standalone)
+                {
+                    SetupSuppresionHandlingInNuGetAnalyzers();
+                }
+                else
+                {
+                    CleanupSuppresionHandlingInNuGetAnalyzers();
+                }
+            }
+            catch (Exception ex)
```

Comment:
Use _Microsoft.VisualStudio.ErrorHandler.IsCriticalException(ex)_ rather than suppressing everything.

Reply:
You mean not critical or critical?

Reply:
Squash non-critical exceptions; allows critical exceptions to propagate.